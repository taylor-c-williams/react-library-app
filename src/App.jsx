import { BrowserRouter as Router, Switch, Route, NavLink } from 'react-router-dom'
import BookList from './views/Books/BookList'
import BookDetail from './views/Books/BookDetail'
import './App.css'

function App() {
  // TODO: Add routes to books & views
  return (
    <main className="container">
      <Router>
        <h1>Library Catalog</h1>
        Check out the &nbsp;
        <NavLink to="/books" exact>
          Book List
        </NavLink>
        <Switch>
          <Route exact path="/books" component={BookList} />
          <Route exact path="/books/:id" component={BookDetail} />
        </Switch>
      </Router>
    </main>
  )
}

export default App
