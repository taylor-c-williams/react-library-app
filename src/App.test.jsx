/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import App from './App'

it('renders a welcome message when signed out', async () => {
  const { container } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  )

  await screen.getByText('Library Catalog')

  expect(container).toMatchSnapshot()
})
