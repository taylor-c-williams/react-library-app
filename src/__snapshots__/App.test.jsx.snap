// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a welcome message when signed out 1`] = `
<div>
  <main
    class="container"
  >
    <h1>
      Library Catalog
    </h1>
    Check out the Â 
    <a
      href="/books"
    >
      Book List
    </a>
  </main>
</div>
`;
